localstack/utils/bootstrap.py

```diff
@@ -221,17 +221,49 @@ def resolve_apis(services: Iterable[str]) -> Set[str]:
 @functools.lru_cache()
 def get_enabled_apis() -> Set[str]:
     """
-    Returns the list of APIs that are enabled through the SERVICES variable. If the SERVICES variable is empty,
-    then it will return all available services. Meta-services like "serverless" or "cognito", and dependencies are
-    resolved.
+    Returns the list of APIs that are enabled through the combination of the SERVICES variable and
+    STRICT_SERVICE_LOADING variable. If the SERVICES variable is empty, then it will return all available services.
+    Meta-services like "serverless" or "cognito", and dependencies are resolved.
 
     The result is cached, so it's safe to call. Clear the cache with get_enabled_apis.cache_clear().
     """
     services_env = os.environ.get("SERVICES", "").strip()
     services = None
-    if services_env and not is_env_true("EAGER_SERVICE_LOADING"):
-        LOG.warning("SERVICES variable is ignored if EAGER_SERVICE_LOADING=0.")
-    elif services_env:
+
+    if services_env and is_env_true("STRICT_SERVICE_LOADING"):
+        from localstack.services.plugins import SERVICE_PLUGINS
+
+        # SERVICES and STRICT_SERVICE_LOADING are set
+        # we check with SERVICE_PLUGINS.list_available() to cross the user-provided list with the available ones
+        available_services = SERVICE_PLUGINS.list_available()
+        services = [service for service in services_env.split(",") if service in available_services]
+        # TODO: log a message if a service was not supported? see with pro loading
```

Comment:
nit: I think this would be a good idea :)

Reply:
I agree, but I think this might create some issues, I'll need to test it properly, as it might be called once with community, then the cache is cleared and once again in Pro. If the services are not there yet, we might log some messages twice. It will need some local testing but I'll keep that in mind for a following PR.